### ISSUE TITLE
BTD Invalid Particle

### ISSUE THREAD

**Issue by johnson452** [ECP-WarpX/WarpX#4260]
Short description:
I'm trying to do a boosted, 1D, PIC simulation. My simulation crashes after a few hundred time-steps with an invalid particle error which calls a MPI abort. 

Extra info:
It works as intended without the Boosted frame options + BTD. And turning off "BTD" and just having the "Full" diagnostics also works, so the issue seems to be specifically with the BTD. (My system is an M1 MacBook, and everything is build via Homebrew with Apple Clang version 14.0.3 )

Building WarpX:
I've built WarpX with:
`cmake -S . -B build_debug -DCMAKE_BUILD_TYPE=Debug -DWarpX_DIMS="1"; cmake --build build_debug -j10`

Files:
[inputs_1d.txt](https://github.com/ECP-WarpX/WarpX/files/12501610/inputs_1d.txt)
[Backtrace.txt](https://github.com/ECP-WarpX/WarpX/files/12501619/Backtrace.txt)
[Terminal_output.txt](https://github.com/ECP-WarpX/WarpX/files/12501620/Terminal_output.txt)
[warpx_used_inputs.txt](https://github.com/ECP-WarpX/WarpX/files/12501621/warpx_used_inputs.txt)

**Comment by RevathiJambunathan** [comment]
@johnson452 
Thank you for reporting this issue. 
In the BTD part of the inputs, could you please turn off species, i.e., `diag1.write_species = 0` 
and test that, so we isolate the issue to be related to particles only?

Thank you.

**Comment by RemiLehe** [comment]
@RevathiJambunathan 
The BackTrace is a bit sparse, but my guess is that the crash is happening here ; is that your understanding too?
https://github.com/ECP-WarpX/WarpX/blob/development/Source/Diagnostics/BTDiagnostics.cpp#L1112

**Comment by RevathiJambunathan** [comment]
@RemiLehe yes, that is my understanding from the Backtrace too

**Comment by johnson452** [comment]
@RevathiJambunathan 
Thanks for the followup! Running with `diag1.write_species = 0` the simulation proceeded without crashing.

@RemiLehe 
The simulation results are a bit surprising. I attached the last frame. I'm not sure if this is connected to the error or not.

![Figure 0-143](https://github.com/ECP-WarpX/WarpX/assets/69021085/9d10ee77-a95d-45cd-bd34-0a7ce8bf1e1c)
![Figure 0-142](https://github.com/ECP-WarpX/WarpX/assets/69021085/5dfceb28-475e-43fd-bebe-b5e2e96c054e)
![Figure 0-141](https://github.com/ECP-WarpX/WarpX/assets/69021085/4cd7af61-e7ad-4079-8775-78a832acbcc4)
![Figure 0-140](https://github.com/ECP-WarpX/WarpX/assets/69021085/e2f3c3dd-770c-4155-aacc-73f1858b755b)

**Comment by RemiLehe** [comment]
@johnson452 
Thanks! Could you show the plot in the boosted frame (i.e. using the `Full` diagnostics instead of the `BackTransformedDiagnostic`), just before the code crashes?

**Comment by johnson452** [comment]
@RemiLehe 
Yep, here it is (at iteration 275). I've included some zoomed-in panels on the structure.
![Figure 0-157](https://github.com/ECP-WarpX/WarpX/assets/69021085/4c18fc98-8f16-4aa7-94d9-78b63ca5d81c)
![Figure 0-156](https://github.com/ECP-WarpX/WarpX/assets/69021085/64366cc7-e072-4b3e-be3f-eab93b75980b)
![Figure 0-155](https://github.com/ECP-WarpX/WarpX/assets/69021085/58906354-893b-40b4-b2af-fc81c789340a)
![Figure 0-154](https://github.com/ECP-WarpX/WarpX/assets/69021085/5e4ced66-1b77-4e72-a12e-a6a5012f066e)
![Figure 0-153](https://github.com/ECP-WarpX/WarpX/assets/69021085/7a31428d-95d2-4248-8a98-d9c3bef4a6b5)
![Figure 0-152](https://github.com/ECP-WarpX/WarpX/assets/69021085/99f28c45-ea6b-4340-ace1-7cd63a4d13e0)

**Comment by johnson452** [comment]
Reminder: post the full (non-backtransformed) diagnostics at the end of the simulation.

**Comment by johnson452** [comment]
@RemiLehe 
Here are the full diagnostics at time steps 7000, and at the end, 20000:
![Figure 0-165](https://github.com/ECP-WarpX/WarpX/assets/69021085/30da0e28-bbcd-4b70-93ce-b054d6e9af75)
![Figure 0-164](https://github.com/ECP-WarpX/WarpX/assets/69021085/186c57eb-8876-496a-ac9b-db106eabcfb2)
![Figure 0-163](https://github.com/ECP-WarpX/WarpX/assets/69021085/11a823a8-9a04-4837-8a87-a8975b012d43)
![Figure 0-162](https://github.com/ECP-WarpX/WarpX/assets/69021085/4c3a98c4-6f51-4b90-bc81-873fd29f3b96)
![Figure 0-161](https://github.com/ECP-WarpX/WarpX/assets/69021085/4253dc2e-bd93-473c-9a5d-d4095eca4763)
![Figure 0-160](https://github.com/ECP-WarpX/WarpX/assets/69021085/5cc58f2f-2d02-4933-8e48-fb3440468578)
